events/janus_rabbitmqevh.c

```diff
@@ -377,7 +365,20 @@ int janus_rabbitmqevh_init(const char *config_path) {
 		g_free((char *)ssl_key_file);
 	if(config)
 		janus_config_destroy(config);
-	return -1;
+	if(!success) {
+		return -1;
+	}
+	JANUS_LOG(LOG_INFO, "%s initialized!\n", JANUS_RABBITMQEVH_NAME);
+	return 0;
+
+error:
+	/* If we got here, something went wrong */
+	success = FALSE;
+	if(route_key)
+		g_free((char *)route_key);
+	if(exchange)
+		g_free((char *)exchange);
+	goto cleanup_and_return;
```

Comment:
Why put `cleanup_and_return` *before* `error`, and then go backwards? It's probably better to put `error` first and let it flow in `cleanup_and_return` instead; successful runs would have a `goto cleanup_and_return` instead. This way, everything always goes in the same direction, which is more intuitive. This is how we do it in a few other plugins as well, I believe.

Reply:
PS: besides, something like `done` is probably a better name than `cleanup_and_return`, after the reordering.

Reply:
As for reordering, I think the successful branch should have less goto's. I'll do reorder since you have another opinion.